engine/crates/runtime/src/fetch.rs

```diff
@@ -23,6 +23,8 @@ pub struct FetchRequest<'a> {
     pub url: &'a url::Url,
     pub headers: http::HeaderMap,
     pub json_body: String,
+    pub subgraph_name: &'a str,
```

Comment:
Wouldn't add the subgraph_name IMHO

Reply:
Yeah we can handle that in the caller and avoid passing the subgraph name, if that feels better. Just need to add an error variant.

Reply:
nah just leave it as is for now, we'll see later 🤷